File: python/program/dealer.py
from mobil import Mobil

class Dealer:
    def __init__(self, NIB="", nama="", alamat="", listMobil=None):
        self.NIB = NIB
        self.nama = nama
        self.alamat = alamat
        if listMobil is None:
            self.listMobil = []
        else:
            self.listMobil = listMobil

    def getNIB(self):
        return self.NIB

    def getNama(self):
        return self.nama

    def getAlamat(self):
        return self.alamat

    def getListMobil(self):
        return self.listMobil

    def setNIB(self, NIB):
        self.NIB = NIB

    def setNama(self, nama):
        self.nama = nama

    def setAlamat(self, alamat):
        self.alamat = alamat

    def setListMobil(self, listMobil):
        self.listMobil = listMobil

    def tambahMobil(self, mobil: Mobil):
        self.listMobil.append(mobil)

    def tampilkan_mobil(self):
        if not self.listMobil:  # atau: if len(self.listMobil) == 0:
            print("Tidak ada mobil di dealer.")
            return
        
        for mobil in self.listMobil:
            mobil.tampilkan_info()
            print("-----------------------------")

File: java/program/Dealer.java
import java.util.ArrayList;

class Dealer {
    private String NIB;
    private String nama;
    private String alamat;
    private ArrayList<Mobil> listMobil;

    public Dealer() {
        this.NIB = "";
        this.nama = "";
        this.alamat = "";
        this.listMobil = new ArrayList<>();
    }

    public Dealer(String NIB, String nama, String alamat) {
        this.NIB = NIB;
        this.nama = nama;
        this.alamat = alamat;
        this.listMobil = new ArrayList<>();
    }

    public String getNIB() {
        return NIB;
    }

    public String getNama() {
        return nama;
    }

    public String getAlamat() {
        return alamat;
    }

    public ArrayList<Mobil> getListMobil() {
        return listMobil;
    }

    public void setNIB(String NIB) {
        this.NIB = NIB;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public void setAlamat(String alamat) {
        this.alamat = alamat;
    }

    public void tambahMobil(Mobil mobil) {
        this.listMobil.add(mobil);
    }

    public void tampilkanMobil() {
        if(listMobil.isEmpty()) {
            System.out.println("Tidak ada mobil di dealer.");
            return;
        }
        for(Mobil mobil : listMobil) {
            mobil.tampilkanInfo();
            System.out.println("----------------------------------------");
        }
    }
}

